js/ui/appDisplay.js

```diff
@@ -609,8 +720,28 @@ const AllView = new Lang.Class({
                                           // and set it to edit mode
                                           this._addedFolderId = id;
                                       }));
+    },
 
-        this._loadApps();
+    loadGrid: function() {
+        this._maybeAddAppCenterIcon();
+        this.parent();
+    },
+
+    getLayoutIds: function() {
+        let layoutIds = this.parent();
+
+        if (!global.settings.get_boolean(EOS_ENABLE_APP_CENTER_KEY))
+            return layoutIds;
+
+        let appSys = Shell.AppSystem.get_default();
+        if (appSys.lookup_app(EOS_APP_CENTER_ID)) {
```

Comment:
I'd remove this check here tbh. If gnome-software somehow isn't installed, we have bigger problems already...

Reply:
I think it's a good idea to leave this check, since the App Center might not be installed in some images such as eosinstaller (see https://github.com/endlessm/gnome-shell/commit/1140d786a40085807da48e73f04714df2e83ca2b#diff-7dfb7548c6314bf99dd14ab6cb89bc5a)